src/snowflake/cli/api/commands/flags.py

```diff
@@ -13,6 +14,106 @@
 _CLI_BEHAVIOUR = "Global configuration"
 
 
+class OverrideableOption:
```

Comment:
This class needs more tests, has a lot of ifs which should be tested.

Reply:
good point, I've added more tests to test_flags but let me know if there's any cases I haven't thought of